### DISCUSSION TITLE
complie issue

### DISCUSSION THREAD

**Discussion by KusuLeung** [IchthysMaranatha/asterisk-chan-quectel#50] [Q&A]
I want to compile with Openwrt 19.07 and Asterisk 16 for EC20，When this step with " ./configure --with-astversion=16" is reached, it fails.  

========================================
./configure --with-astversion=16
checking build system type... mips-unknown-linux-musl
checking host system type... mips-unknown-linux-musl
checking target system type... mips-unknown-linux-musl
checking for a BSD-compatible install... ./install-sh -c
checking whether build environment is sane... yes
checking for a thread-safe mkdir -p... ./install-sh -c -d
checking for gawk... no
checking for mawk... no
checking for nawk... no
checking for awk... awk
checking whether make sets $(MAKE)... yes
checking whether make supports nested variables... yes
checking for gcc... gcc
checking whether the C compiler works... yes
checking for C compiler default output file name... a.out
checking for suffix of executables... 
checking whether we are cross compiling... no
checking for suffix of object files... o
checking whether we are using the GNU C compiler... yes
checking whether gcc accepts -g... yes
checking for gcc option to accept ISO C89... none needed
checking whether gcc understands -c and -o together... yes
checking for style of include used by make... GNU
checking dependency style of gcc... none
checking how to run the C preprocessor... gcc -E
checking for strip... strip
checking for rm... rm
checking for library containing iconv... none required
checking for sqlite3_open in -lsqlite3... no
configure: error: sqlite3 library missing

====================================
opkg install libsqlite3
Package libsqlite3-0 (3310100-1) installed in root is up to date.

And I have installed sqlite3, but it doesn't work.
Thanks for help.

**Comment by IchthysMaranatha** [comment]
You need to use the patches correctly. Patch dir has to be on same level as makefile. Anyway, you can use already compiled binary here https://github.com/IchthysMaranatha/asterisk-chan-quectel/releases/tag/3.5 if your hardware is mips 24kc compatible

**Reply by KusuLeung** [reply]
Thanks for reply. 
But already I have compiled the tag 3.5，it still doesn't work, and sqlite3 library is not found.
Could you built it for me？

**Comment by IchthysMaranatha** [comment]
Which hardware are you compiling for? Did you try and use the binary already linked?

**Reply by KusuLeung** [reply]
Openwrt 19.07 mips 24kc and Asterisk 16.25 for EC20.
I try many times，config still stuck sqlite3.

**Reply by IchthysMaranatha** [reply]
Use binary, no need to compile, are you not understanding this?

**Reply by IchthysMaranatha** [reply]
Compiling is to make the executable, use executable already compiled, just download quectel.so and put it in the asterisk modules folder, usually /usr/lib/asterisk/modules/. 
Place quectel conf in /etc/asterisk/ folder. Modify conf as needed and restart asterisk.

**Reply by KusuLeung** [reply]
I tried using quectel.so, it didn't work. So I want to generate the binary file again through compilation.
This issue has been bothering me for a long time. I can't find any other solution.


***********************************************************
~:/usr/lib/asterisk/modules# ls -ltr chan_quectel.so
-rwxr-xr-x    1 root     root        225275 Aug 17 01:28 chan_quectel.so

~# ls -ltr /etc/asterisk/quectel.conf 
-rw-r--r--    1 root     root          4544 Aug 17 07:10 /etc/asterisk/quectel.conf

~# asterisk -rvvv
Asterisk 16.25.2, Copyright (C) 1999 - 2021, Sangoma Technologies Corporation and others.
Created by Mark Spencer <markster@digium.com>
Asterisk comes with ABSOLUTELY NO WARRANTY; type 'core show warranty' for details.
This is free software, with components licensed under the GNU General Public
License version 2 and other licenses; you are welcome to redistribute it under
certain conditions. Type 'core show license' for details.

Connected to Asterisk 16.25.2 currently running on GL-XE300 (pid = 11979)
GL-XE300*CLI> quectel show devices
No such command 'quectel show devices' (type 'core show help quectel show' for other possible commands)

**Comment by IchthysMaranatha** [comment]
What happens with 
module show like chan_quectel
Is alsa-lib installed?
You need to trouble shoot as to why module is not properly loaded and running

**Comment by IchthysMaranatha** [comment]
Check log files and see why it's not running and resolve accordingly. You already have the correct binary, no need for compiling anything again. If you have copied with winscp, you need to copy in binary mode

**Reply by KusuLeung** [reply]
[Aug 17 09:18:44] ERROR[29352] loader.c: Error loading module 'chan_quectel.so': Error loading shared library /usr/lib/asterisk/modules/chan_quectel.so: Exec format error

the log file displays this error.

**Reply by IchthysMaranatha** [reply]
Then it's not the proper executable, either not copied correctly or not correct for hw. Again what exactly is your hardware? Which router model is this?

**Reply by IchthysMaranatha** [reply]
If your router is GL-XE300 then it is not mips24kc, you need to compile with sdk for target ath79

**Reply by IchthysMaranatha** [reply]
You can compile correctly if you place the patches folder where your make file is when creating custom feed with sdk

**Reply by KusuLeung** [reply]
I downloaded the chan_quectel.so directly in GL-XE300 router, it should be no problem.
Or could you give me the CRC for checking？Thanks.


~# md5sum chan_quectel.so
106d06f3ba24cec9b6b8405b8f00142e  chan_quectel.so

~# sha512sum chan_quectel.so
a09dfe7bedfb9947afd15cfbaa94d96f9152671a46a62fef7ea3dada4a9cbc260a6e28b5c2f0d9d3b2179647aa0f4c86acd241a5c04755cc52c9e81b5b9a3163  chan_quectel.so

**Reply by KusuLeung** [reply]
~# uname -a
Linux GL-XE300 4.14.241 #0 Thu Jul 29 19:50:28 2021 mips GNU/Linux

I used to think it is mips 24kc.

**Reply by IchthysMaranatha** [reply]
Is this not your router
https://openwrt.org/toh/gl.inet/gl-xe300 ? Or does the same model come with diff architectures?

**Comment by IchthysMaranatha** [comment]
I think you'll need to compile it with ath79 sdk not the ramips one, I've compiled with because of the QCA9531 soc

**Comment by IchthysMaranatha** [comment]
This is the executable working on my mips24kc (we826) router with openwrt 19.07, have zipped it to attach it
[chan_quectel.zip](https://github.com/IchthysMaranatha/asterisk-chan-quectel/files/12369211/chan_quectel.zip)
